### PR TITLE
CVS-46541 Implement Sequence class

### PR DESCRIPTION
**Pull request by mzegla**

### CODE REVIEW

File: src/sequence.cpp [hunk 1]
```diff
@@ -0,0 +1,58 @@
+//*****************************************************************************
+// Copyright 2020 Intel Corporation
```
Comment (rasapala) [inline src/sequence.cpp:2]: 2021

File: src/sequence.hpp [hunk 2]
```diff
@@ -0,0 +1,52 @@
+//*****************************************************************************
+// Copyright 2020 Intel Corporation
```
Comment (rasapala) [inline src/sequence.hpp:2]: 2021

**Review by michalkulakowski — APPROVED** [review]

File: src/test/sequence_test.cpp [hunk 3]
```diff
@@ -0,0 +1,33 @@
...
+//*****************************************************************************
+#include <memory>
+#include <mutex>
+
+#include <gmock/gmock.h>
+#include <gtest/gtest.h>
+
+#include "../sequence.hpp"
+
+TEST(Sequence, MovedMutexNullified) {
```
Comment (rasapala) [inline src/test/sequence_test.cpp:24]: I think adding ULT for  updateMemoryState is crucial here. Also we should check last modification time was updated.

File: src/test/sequence_test.cpp [hunk 4]
```diff
@@ -0,0 +1,130 @@
...
+    std::vector<float> state1(elementsCount1);
+    std::iota(state1.begin(), state1.end(), 0);
+    addState(newState, "state1", shape1, state1);
+
+    ovms::Sequence sequence;
+    auto time1 = sequence.getLastActivityTime();
+    std::this_thread::sleep_for(std::chrono::seconds(1));
+    sequence.updateMemoryState(newState);
+    auto time2 = sequence.getLastActivityTime();
+    EXPECT_NE(time1, time2);
```
Comment (rasapala) [inline src/test/sequence_test.cpp:98]: Check grather than

**Review by rasapala — APPROVED** [review]
LGTM

**Pull request merged**